sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/grouping.scala

```diff
@@ -26,7 +26,7 @@ import org.apache.spark.sql.types._
 /**
  * A placeholder expression for cube/rollup, which will be replaced by analyzer
  */
-trait GroupingSet extends Expression with CodegenFallback {
+trait GroupingAnalytic extends Expression with CodegenFallback {
```

Comment:
Do you mean `GroupingAnalytics`? and does it represent all grouping analysis including group-bys?

Reply:
another option is `BaseGroupingSets`. cube/rollup is syntax sugar for grouping sets.

Reply:
> BaseGroupingSets

Change to this. Seems the scope of expression `GroupingAnalytics` is too large.